spec/requests/user_actions_controller_spec.rb

```diff
@@ -99,5 +99,40 @@
       end
     end
 
+    context "other users' activity" do
+      let(:another_user) { Fabricate(:user) }
+
+      UserAction.private_types.each do |action_type|
+        action_name = UserAction.types.key(action_type)
+        it "users cannot list other users' actions of type: #{action_name}" do
```

Comment:
Do we need to sign in a user here? Otherwise, this is actually testing that an anon user cannot see the route rather than another user cannot view the route.

Reply:
> Do we need to sign in a user here?

Yeah, we need it, thank you for catching it. 

But we actually need to check anonymous users too, so I've added more test cases.